src/java_bytecode/java_object_factory.cpp

```diff
@@ -841,20 +841,19 @@ void java_object_factoryt::gen_nondet_pointer_init(
   // and asign to `expr` the address of such object
   code_blockt non_null_inst;
 
-  if(
-    java_string_library_preprocesst::implements_java_char_sequence_pointer(
-      expr.type()))
-  {
-    add_nondet_string_pointer_initialization(
-      expr,
-      object_factory_parameters.max_nondet_string_length,
-      object_factory_parameters.string_printable,
-      symbol_table,
-      loc,
-      object_factory_parameters.function_id,
-      assignments);
-  }
-  else
+  // Note string-type-specific initialization might fail, e.g. if java.lang.CharSequence does not
```

Comment:
I'm not a big fan of using critical side effect functions in the if statement as they tend to get lost. I think it would be clearer to:

```C++
if(!java_string_library_preprocesst::implements_java_char_sequence_pointer(
  expr.type()))
{
  bool error_adding_string_pointer = add_nondet_string_pointer_initialization(...);
  if(error_adding_string_pointer)
  {
    // tring-type-specific initialization failed, e.g. if java.lang.CharSequence does not ...
    gen_pointer_target_init(...);
  }
}
else 
{
  get_pointer_target_init(...);
}
```